### PR TITLE
Update Danger and use scoped syntax for git/github metadata

### PR DESCRIPTION
**Pull request by orta**
http://danger.systems now shows the CocoaPods Dangerfile as an example on the front page, so I wanted to to make sure it lives up the the Danger Dream.

### CODE REVIEW

File: Dangerfile [hunk 1]
```diff
@@ -10,14 +10,14 @@
...
   fail('Please rebase to get rid of the merge commits in this PR')
 end
 
 # Request a CHANGELOG entry, and give an example
-has_app_changes = !modified_files.grep(/lib/).empty?
-if !modified_files.include?('CHANGELOG.md') && has_app_changes
-  fail('Please include a CHANGELOG entry to credit yourself! \nYou can find it at [CHANGELOG.md](https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md).', :sticky => false)
+has_app_changes = !git.modified_files.grep(/lib/).empty?
+if !git.modified_files.include?('CHANGELOG.md') && has_app_changes
+  fail("Please include a CHANGELOG entry to credit yourself! \nYou can find it at [CHANGELOG.md](https://github.com/CocoaPods/CocoaPods/blob/master/CHANGELOG.md).", :sticky => false)
```
Comment (orta) [inline Dangerfile]: This is re: https://github.com/danger/danger/issues/321

**Comment by segiddins** [comment]
👍

**Comment by orta** [comment]
OK, two of these builds failed on different things, and one passed 👯 

Fails:
screen shot 2016-07-18 at 3 47 29 pm

screen shot 2016-07-18 at 3 47 47 pm

However the other one passes, I've hit retry on the failed ones.

**Comment by DanToml** [comment]
@orta One of them is a transient failure, and the other is fixed by #5640, so this LGTM

**Comment by orta** [comment]
OK, let's do it.

**Pull request merged**